asciidoctorj-core/src/test/java/org/asciidoctor/WhenAsciidoctorLogsToConsole.java

```diff
@@ -255,6 +255,16 @@ public void shouldNotifyLogHandlerService() {
         }
     }
 
+    @Test
+    public void shouldCreateNewLogHandlersOnCreateAsciidoctor() {
```

Comment:
Nitpicking here, but this test is not isolated. I see other Asciidoctor instances, at least one was created by `AsciidoctorExtension`.

I'd rather run its own test class, that way we don't need to do extra inits, we can make the counter private and accessible via getter `getInstancesCount`.

```
class WhenAsciidoctorLogsToConsoleIsolated {

    @Test
    void shouldCreateNewLogHandlersOnCreateAsciidoctor() {
        Asciidoctor asciidoctor = Asciidoctor.Factory.create();
        asciidoctor.convert("Test", Options.builder().build());
        Asciidoctor asciidoctor2 = Asciidoctor.Factory.create();
        asciidoctor2.convert("Test", Options.builder().build());

        assertEquals(2, TestLogHandlerService.getInstancesCount());
    }

}
```

Reply:
My concern is that if we run tests in parallel, this tests could fail.

Reply:
> If we run the test methods run in parallel, then we could indeed see isolation problems, but we would have the same issues if we run the test classes in parallel, no?

If you mean without classpath isolation indeed, I haven't thought of all the complexities tbh.
The Gradle configuration seems a but too much for a single test, I wonder if this could be used https://junit.org/junit5/docs/snapshot/user-guide/#launcher-api-launcher-interceptors-custom, but I am afraid creating a new classloader with the same dependencies may be also very painful. Whatever is simpler

Reply:
I moved the test into a separate class and set the execution mode to SAME_THREAD.
If I understand https://junit.org/junit5/docs/snapshot/user-guide/#writing-tests-parallel-execution correctly this should guarantee that no other test would run in parallel in the same JVM.

Reply:
That said, the diagram in the documentation makes me believe I did not understand it correctly. :(

Reply:
Let's call it fine for now :D